modules/exploits/unix/webapp/spip_rce_form.rb

```diff
@@ -0,0 +1,157 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'SPIP form PHP Injection',
+        'Description' => %q{
+          This module exploits a PHP code injection in SPIP. The vulnerability exists in the
+          oubli parameter and allows an unauthenticated user to execute arbitrary commands
+          with web user privileges. Branches 3.2, 4.0, 4.1 and 4.2 are concerned. Vulnerable versions
+          are <3.2.18, <4.0.10, <4.1.18 and <4.2.1.
+        },
+        'Author' => [
+          'coiffeur',       # Initial discovery
+          'Laluka',         # PoC
+          'Julien Voisin'   # MSF module
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          [ 'URL', 'https://blog.spip.net/Mise-a-jour-critique-de-securite-sortie-de-SPIP-4-2-1-SPIP-4-1-8-SPIP-4-0-10-et.html' ],
+          [ 'URL', 'https://therealcoiffeur.com/c11010' ],
```

Comment:
https://therealcoiffeur.com/c11010 is unfortunately returning a 502 for me atm. Was this a link to the PoC and do you think it's possible they've taken it down as per someone's request?
```suggestion
          [ 'URL', 'https://blog.spip.net/Mise-a-jour-critique-de-securite-sortie-de-SPIP-4-2-1-SPIP-4-1-8-SPIP-4-0-10-et.html' ]
```

Reply:
It's down temporarily, it should come back any time soon.